src/libopensc/pkcs15.c

```diff
@@ -2465,6 +2465,10 @@ sc_pkcs15_read_file(struct sc_pkcs15_card *p15card, const struct sc_path *in_pat
 			sc_pkcs15_cache_file(p15card, in_path, data, len);
 		}
 	}
+	if (len == 0) {
+		free(data);
+		data = NULL;
+	}
```

Comment:
I would keep this condition in the above branch, as the `data` is only allocated there and we already have a condition for caching results in there

Reply:
moved to the branch above